patches/kex_rlwe_newhope_avx2/patch1.patch

```diff
@@ -0,0 +1,1073 @@
+ }
+ 
+ 
+-void newhope_shareda(unsigned char *sharedkey, const poly *sk, const unsigned char *received)
++static void newhope_shareda(unsigned char *sharedkey, const poly *sk, const unsigned char *received)
+ {
+   poly v,bp, c;
+ 
+   decode_b(&bp, &c, received);
+ 
+-  poly_pointwise(&v,sk,&bp);
++  OQS_poly_pointwise(&v,sk,&bp);
+   poly_invntt(&v);
+  
+-  rec(sharedkey, &v, &c);
++  OQS_rec(sharedkey, &v, &c);
+ 
+ #ifndef STATISTICAL_TEST 
+   sha3256(sharedkey, sharedkey, 32); 
+diff --git a/src/kex_rlwe_newhope/avx2/ntt.h b/src/kex_rlwe_newhope/avx2/ntt.h
+index c24b137..1cb540a 100644
+--- a/src/kex_rlwe_newhope/avx2/ntt.h
++++ b/src/kex_rlwe_newhope/avx2/ntt.h
+@@ -8,8 +8,8 @@ extern int32_t psis_inv[];
+ extern double omegas_double[];
+ extern double omegas_inv_double[];
+ 
+-void bitrev_vector(int32_t* poly);
+-void pwmul_double(int32_t* poly, const int32_t* factors);
+-void ntt_double(int32_t*,const double*,const double*);
++void OQS_bitrev_vector(int32_t* poly);
++void OQS_pwmul_double(int32_t* poly, const int32_t* factors);
++void OQS_ntt_double(int32_t*,const double*,const double*);
+ 
+ #endif
+diff --git a/src/kex_rlwe_newhope/avx2/ntt_double.s b/src/kex_rlwe_newhope/avx2/ntt_double.s
+index 1ec429f..d876550 100644
+--- a/src/kex_rlwe_newhope/avx2/ntt_double.s
++++ b/src/kex_rlwe_newhope/avx2/ntt_double.s
+@@ -76,9 +76,9 @@
+ # qhasm: enter ntt_double
+ .p2align 5
+ .global _ntt_double
+-.global ntt_double
++.global OQS_ntt_double
+ _ntt_double:
+-ntt_double:
++OQS_ntt_double:
+ mov %rsp,%r11
+ and $31,%r11
+ add $0,%r11
+diff --git a/src/kex_rlwe_newhope/avx2/poly.c b/src/kex_rlwe_newhope/avx2/poly.c
+index 43c0df2..44b0956 100644
+--- a/src/kex_rlwe_newhope/avx2/poly.c
++++ b/src/kex_rlwe_newhope/avx2/poly.c
+@@ -1,7 +1,7 @@
+ #include "poly.h"
+ #include "ntt.h"
+-#include "randombytes.h"
+-#include "fips202.h"
++#include "randombytes.c"
++#include "fips202.c"
+ #include "crypto_stream.h"
+ 
+ static uint16_t barrett_reduce(uint16_t a)
+@@ -14,7 +14,7 @@ static uint16_t barrett_reduce(uint16_t a)
+   return a;
+ }
+ 
+-void poly_frombytes(poly *r, const unsigned char *a)
++static void poly_frombytes(poly *r, const unsigned char *a)
+ {
+   int i;
+   for(i=0;i<PARAM_N/4;i++)
+@@ -26,7 +26,7 @@ void poly_frombytes(poly *r, const unsigned char *a)
+   }
+ }
+ 
+-void poly_tobytes(unsigned char *r, const poly *p)
++static void poly_tobytes(unsigned char *r, const poly *p)
+ {
+   int i;
+   uint16_t t0,t1,t2,t3,m;
+@@ -70,7 +70,7 @@ void poly_tobytes(unsigned char *r, const poly *p)
+ 
+ 
+ 
+-void poly_uniform(poly *a, const unsigned char *seed)
++static void poly_uniform(poly *a, const unsigned char *seed)
+ {
+   unsigned int pos=0, ctr=0;
+   uint16_t val;
+@@ -99,9 +99,9 @@ void poly_uniform(poly *a, const unsigned char *seed)
+ }
+ 
+ 
+-extern void cbd(poly *r, unsigned char *b);
++extern void OQS_cbd(poly *r, unsigned char *b);

```

Comment:
Should that be `static` (after including the c file rather than renaming the function)?

Reply:
Oh, is it because it is in assembly?

Reply:
Yes.